File: Trabajo python/ejercicio_12.py
import os
os.system("cls")


numero = int(input("Numero: "))

for posicion in [1, 2]:
    resultado = numero << posicion

    factor = 2 if posicion == 1 else 4
    print(f"Multiplicado por {factor} desplazando {posicion} bits: {resultado}")

File: Trabajo js/ejercicio_12.js
let numero = 7;

while(numero !== 1)
{
    console.log(numero);

    if(numero % 2 === 0)
    {
        numero = numero / 2;
    }
    else
    {
        numero = (numero * 3) + 1;
    }
}
console.log(numero);